pkg/controllers/provisioningv2/rke2/machinestatus/machinestatus.go

```diff
@@ -266,7 +275,7 @@ func (h *handler) getInfraMachineState(capiMachine *capi.Machine) (status corev1
 	machine, err := h.dynamic.Get(gvk, capiMachine.Namespace, capiMachine.Spec.InfrastructureRef.Name)
 	if apierror.IsNotFound(err) {
 		if capiMachine.DeletionTimestamp != nil {
-			return corev1.ConditionUnknown, "MachineDeleted", "machine is being deleted", "", nil
+			return corev1.ConditionFalse, "Deleted", "machine infrastructure is deleted", "", err
```

Comment:
Per offline, attempt to use the upstream CAPI consts here

Reply:
Will use the `MachineDelete` reason constant.